dictionary-builder/src/main.rs

```diff
@@ -44,409 +37,253 @@ fn main() {
-            format!("Group100(Tag(0x{}00, 0x{}))", group, elem)
+            tag_type = TagType::Group100;
+            format!("Tag(0x{}00, 0x{})", group, elem)
         } else if let Some(cap) = regex_tag_element100.captures(tag.as_str()) {
             // tag range over elements: (gggg, eexx)
             let group = cap.get(1).expect("capture group 1: group").as_str();
             let elem = cap
                 .get(2)
                 .expect("capture group 2: element portion")
                 .as_str();
-            format!("Element100(Tag(0x{}, 0x{}00))", group, elem)
+            tag_type = TagType::Element100;
+            format!("Tag(0x{}, 0x{}00)", group, elem)
         } else {
+            panic!("invalid tag: {}", alias);
+        };
+
+        result.push(Entry {
+            tag,
+            vr,
+            alias,
+            vm: parts[3].to_string(),
+            obs: parts[4].to_string(),
+            is_retired: parts[4].contains("retired"),
+            tag_declaration,
+            tag_type,
+        });
+    }
+
+    Ok(result)
+}
+
+#[derive(Debug, PartialEq, Eq, PartialOrd, Clone, Serialize)]
+enum TagType {
+    Single,
+    Group100,
+    Element100,
+}
+
+#[derive(Debug, PartialEq, Eq, PartialOrd, Clone, Serialize)]
+struct Entry {
+    /// (0010,0010)
+    tag: String,
+    /// PN
+    vr: String,
+    /// PatientName
+    alias: String,
+    /// 1
+    vm: String,
+    /// DICOM
+    obs: String,
+    /// Retired field?
+    is_retired: bool,
+    /// Tag declaration "Tag(0x6000, 0x1102)"
+    tag_declaration: String,
+    /// The type the tag represents
+    tag_type: TagType,
+}
+
+fn to_code_file<P: AsRef<Path>>(
+    dest_path: P,
+    entries: Vec<Entry>,
+    include_retired: bool,
+) -> DynResult<()> {
+    if let Some(p_dir) = dest_path.as_ref().parent() {
+        create_dir_all(&p_dir)?;
+    }
+    let mut f = File::create(&dest_path)?;
+
+    f.write_all(
+        b"//! Automatically generated. Edit at your own risk.\n\n\
+    use dicom_core::dictionary::{DictionaryEntryRef, TagRange, TagRange::*};\n\
+    use dicom_core::Tag;\n\
+    use dicom_core::VR::*;\n\n",
+    )?;
+
+    for e in &entries {
+        if !include_retired && e.is_retired {
             continue;
+        }
+
+        writeln!(f, "/// {} {} {} {} {}", e.alias, e.tag, e.vr, e.vm, e.obs)?;
+
+        let tag_type = match e.tag_type {
+            TagType::Single => "Tag",
+            _ => "TagRange",
+        };
+
+        let tag_set = match e.tag_type {
+            TagType::Single => e.tag_declaration.clone(),
+            TagType::Group100 => format!("Group100({})", e.tag_declaration),
+            TagType::Element100 => format!("Element100({})", e.tag_declaration),
         };
 
-        let mut vr = vr.unwrap_or_else(|| "".into());
-        if vr == "See Note" {
-            vr = "UN See Note".to_string();
+        writeln!(
+            f,
+            "#[rustfmt::skip]\npub const {}: {} = {}; // {}",
```

Comment:
Now that we have a doc comment on each declaration, I believe the trailing comment is no longer needed.

```suggestion
            "#[rustfmt::skip]\npub const {}: {} = {};",
```